server/middleware/populateCurrentUser.ts

```diff
@@ -1,24 +1,41 @@
 import { RequestHandler } from 'express'
+import { jwtDecode } from 'jwt-decode'
 import logger from '../../logger'
-import UserService from '../services/userService'
+import { convertToTitleCase } from '../utils/utils'
 
-export default function populateCurrentUser(userService: UserService): RequestHandler {
+export default function populateCurrentUser(): RequestHandler {
   return async (req, res, next) => {
     try {
-      if (res.locals.user) {
-        const user = res.locals.user && (await userService.getUser(res.locals.user.token))
-        if (user) {
-          if (!user.activeCaseLoadId) {
-            user.activeCaseLoadId = await userService.getActiveCaseLoadId(res.locals.user.token)
-          }
-          res.locals.user = { ...user, ...res.locals.user }
-        } else {
-          logger.info('No user available')
-        }
+      const {
+        name,
+        user_id: userId,
+        authorities: roles = [],
+      } = jwtDecode(res.locals.user.token) as {
+        name?: string
+        user_id?: string
+        authorities?: string[]
       }
+
+      res.locals.user = {
+        ...res.locals.user,
+        userId,
+        name,
+        displayName: convertToTitleCase(name),
+        userRoles: roles.map(role => role.substring(role.indexOf('_') + 1)),
+      }
+
+      if (res.locals.user.authSource === 'nomis') {
+        res.locals.user.staffId = parseInt(userId, 10) || undefined
+      }
+
+      if (res.locals.feComponentsMeta) {
+        const { feComponentsMeta } = res.locals
+        res.locals.user.activeCaseLoadId = feComponentsMeta.activeCaseLoad.caseLoadId
+      }
```

Comment:
Updated from template, with just the addition of these lines to populate the user's case load.